src/main/java/com/amihaiemil/versioneye/MkVersionEye.java

```diff
@@ -88,7 +106,29 @@ public VersionEye trusted() throws IOException {
 
     @Override
     public Me me() {
+        if(this.authenticated == null) {
+            throw new IllegalStateException("No Authenticated User.");
+        }
         return new MkMe(this.server, this.authenticated);
     }
+    
+    /**
+     * Add authenticated user to the MkServer.
+     */
+    private void addAuthenticatedUserToServer() {
+        JsonArray array = this.server.storage().build()
+            .getJsonArray("authenticated");
+        if(array == null) {
```

Comment:
@SherifWaly This check here is really ugly. Doesn't ``getJsonArray(...)`` method have the option to specify the default value if it's null? Same as it is with getString(...). E.g. you can say ``JsonObject.getString("test", "')``, to return empty String if no "test" attribute is found :D